pynamodb/attributes.py

```diff
@@ -552,11 +551,12 @@ def __set__(self, instance, value):
         """
         if not instance or self._is_map_attribute_class_object(instance):
             return
+        current_timestamp_s = datetime.utcnow().timestamp()
```

Comment:
I led you down the wrong track here. If we use a `datetime` instance then we'd need to make it tz aware for this to work correctly. Also I don't think that `timestamp()` is available in py2, so `time.time()` is indeed better here.